packages/account-abstraction/src/BaseAccountAPI.ts

```diff
@@ -258,7 +258,6 @@ export abstract class BaseAccountAPI {
 
     // Review: doesn't work if wallet is not already deployed.
     const callGasLimit =
-      parseNumber(detailsForUserOp.gasLimit) ??
```

Comment:
what is the case here at this stage? '// Review: doesn't work if wallet is not already deployed.'

and why this is removed? :  parseNumber(detailsForUserOp.gasLimit) ??

Reply:
it works even wallet is not deployed have tested this

Reply:
naa, it does not work. just check all of my trxs on chain. there was internal failure of adding liquidity to hyphen contracts.

update code to this

```
const callGasLimit = (await this.provider.estimateGas({
          from: this.entryPoint.address,
          to: this.getAccountAddress(),
          data: callData
        }))
        // if wallet is not deployed we need to multiply estimated limit to 3 times to get accurate callGasLimit
        if (!this.isDeployed)
        callGasLimit = callGasLimit.mul(3)
```

Reply:
why 3 times?